### PR TITLE
Update docs for pulumi v3 providers & misc

### PR DESCRIPTION
**Pull request by elsesiy**
Update formatting in workflows
Update to go1.16.3

### CODE REVIEW

**Comment by elsesiy** [comment]
@viveklak Could you please run the workflows & let me know whether this misses anything? Thanks!

**Comment by viveklak** [comment]
@elsesiy could you limit this to just the workflow formatting? I have v3 update etc. coming in https://github.com/pulumi/pulumi-kubernetes-operator/pull/137. Thanks!

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/operator-framework/operator-lib v0.0.0-20200728190837-b76db547798d
 	github.com/operator-framework/operator-sdk v0.19.0
 	github.com/pkg/errors v0.9.1
-	github.com/pulumi/pulumi/sdk/v3 v3.1.1-0.20210428183529-c0b5339cf376
+	github.com/pulumi/pulumi/sdk/v3 v3.1.0
```
Comment (viveklak) [inline go.mod:12]: We want `v3.1.1-0.20210428183529-c0b5339cf376` retained

File: go.mod [hunk 2]
```diff
@@ -20,7 +20,6 @@ require (
 	k8s.io/apimachinery v0.18.4
 	k8s.io/client-go v12.0.0+incompatible
 	sigs.k8s.io/controller-runtime v0.6.1
-	sigs.k8s.io/controller-tools v0.3.0 // indirect
```
Comment (viveklak) [inline go.mod:23]: This is an indirect dependency required for codegen, could you add this back. Sadly this breaks with `go mod tidy`. Need a better way to handle this. Can you make sure to run `make generate-crds` before committing?

File: README.md [hunk 3]
```diff
@@ -657,11 +657,11 @@ class MyStack : Stack
 package main
 
 import (
-	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/apps/v1"
-	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/core/v1"
-	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/meta/v1"
-	rbacv1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/rbac/v1"
-	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/apps/v1"
```
Comment (viveklak) [inline README.md:660]: Thanks for updating these!

File: go.mod (hunk 2 above)
Comment (elsesiy) [reply go.mod]: Done!

**Review by viveklak — APPROVED** [review]

**Comment by viveklak** [comment]
This is blocked on https://github.com/pulumi/pulumi-kubernetes-operator/issues/142. We will merge this as soon as the above is fixed.
Thanks @elsesiy!

**Comment by viveklak** [comment]
@elsesiy we can now handle external PRs on this repo. Did you want to update with merge conflicts resolved?

**Pull request merged**